shops/src/controllers/products.ts

```diff
@@ -0,0 +1,70 @@
+import Product from "../models/products";
+import { reqRes } from "../utils/interfaces";
+import { ProductNotFoundError } from "../utils/errors";
+
+// CREATE
+
+export const postProduct: reqRes = async (req, res) => {
+  try {
+    const product = new Product(req.body);
+    await product.save();
+    res.status(201).send(product);
+  } catch (err) {
+    res.status(400).send("Bad Request");
+  }
+};
+// READ ALL
+
+export const getProducts: reqRes = async (req, res) => {
+  try {
+    const product = await Product.find();
+    res.status(200).send(product);
+  } catch (err) {
+    res.status(500).send("Server Error");
+  }
+};
+
+// READ ONE
+
+export const getProduct: reqRes = async (req, res) => {
+  try {
+    const product = await Product.findById(req.params.id);
+    if (!product) throw new ProductNotFoundError();
+    res.status(200).send(product);
+  } catch (err) {
+    let status = { code: 500, message: "Server Error" };
+    if (err instanceof ProductNotFoundError) {
+      status = { code: 404, message: err.message };
+    }
+    res.status(status.code).send(status.message);
+  }
+};
+
+// UPDATE
+
+export const updateProduct: reqRes = async (req, res) => {
+  try {
+    const product = await Product.findById(req.params.id);
+    if (!product) throw new ProductNotFoundError();
+    if (req.body.name) product.name = req.body.name;
+    if (req.body.price) product.price = req.body.product;
```

Comment:
You need .price instead of .product on this line. We are replacing the old price with the new price sent in the body of the request. 

```    if (req.body.price) product.price = req.body.price;```

I would also like to see the photoURL included so we could replace images if we wanted to. 

```    if (req.body.photoURL) product.photoURL = req.body.photoURL;```